polaris-service/src/main/java/org/apache/polaris/service/config/RealmEntityManagerFactory.java

```diff
@@ -18,29 +18,39 @@
  */
 package org.apache.polaris.service.config;
 
+import jakarta.inject.Inject;
+import jakarta.inject.Provider;
+import java.util.HashMap;
 import java.util.Map;
-import java.util.concurrent.ConcurrentHashMap;
 import org.apache.polaris.core.context.RealmContext;
+import org.apache.polaris.core.context.RealmScope;
 import org.apache.polaris.core.persistence.MetaStoreManagerFactory;
 import org.apache.polaris.core.persistence.PolarisEntityManager;
+import org.apache.polaris.core.persistence.cache.EntityCache;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /** Gets or creates PolarisEntityManager instances based on config values and RealmContext. */
+@RealmScope
 public class RealmEntityManagerFactory {
```

Comment:
I disagree here, this is clearly an application-scoped bean, holding long-lived instances of `PolarisEntityManager` keyed by realm id.

Reply:
You're right. I'd actually like to get rid of this, as I don't think it adds much with the CDI infra, but I'll keep it as a singleton for now